Area51/Sources/SubredditsViewController.swift

```diff
@@ -16,28 +25,63 @@ class SubredditsViewController: UIViewController {
     private func setupTableView() {
         self.tableView.delegate = self
         self.tableView.dataSource = self
+        self.tableView.refreshControl = UIRefreshControl()
+        self.tableView.refreshControl?.addTarget(self, action: #selector(self.refreshData), for: .valueChanged)
     }
 
     private func showSubreddit(_ subreddit: Subreddit) {
         let feedViewController = FeedViewController.instantiateFromStoryboard(withSubreddit: subreddit)
         self.splitViewController?.showDetailViewController(feedViewController.embeddedInNavigationController(),
                                                            sender: self)
     }
+    @objc
+    private func refreshData() {
+        self.dataSource.refresh()
+    }
 }
 
 extension SubredditsViewController: UITableViewDataSource, UITableViewDelegate {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section {
+        case 0:
+            return "Reddit Feeds"
```

Comment:
Can we define this as an enum somewhere?
```
enum SubredditsSections: Int {
case redditFeeds = 0
...

var title: String {
    switch self {
        case .redditFeeds:
            return "Reddit Feeds"

...

var allSections: [SubredditsSections] {
  return [.redditFeeds, .defaultSubreddits]
```

Then you can change the tableView methods to something like:

```
func numberOfSections(in tableView: UITableView) -> Int {
    return SubredditsSections.allSections.count
}
```

```
func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
   return SubredditsSections(rawValue: section).title
```